libs/Microsoft.MixedReality.WebRTC/ISignalingService.cs

```diff
@@ -0,0 +1,221 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+
+namespace Microsoft.MixedReality.WebRTC
+{
+    public enum SignalingType
+    {
+        SdpOffer,
+        SdpAnswer,
+        IceCandidate
+    }
+
+    public class SignalingMessage
+    {
+        public string SessionId;
+        public SignalingType Type;
+        public string Payload;
+    }
+
+    interface ISignalingService : IDisposable
+    {
+        /// <summary>
+        /// Start the service and make the local process visible as `localEndPointId`.
+        /// </summary>
+        void Start(string localEndPointId);
+
+        /// <summary>
+        /// Stop the service.
+        /// </summary>
+        void Stop();
+
+        /// <summary>
+        /// Get the endpoints that are currently active.
+        /// </summary>
+        IEnumerable<string> RemoteEndPoints { get; }
+
+        /// <summary>
+        /// Raised every time there is a change in the list of remote endpoints.
+        /// Every delegate added to this is called once even if there are no updates.
+        /// </summary>
+        event Action RemoteEndPointsChanged;
+
+        /// <summary>
+        /// Send a message to the remote endpoint.
+        /// </summary>
+        void SendToRemoteEndpoint(string targetEndPointId, SignalingMessage message);
+
+        /// <summary>
+        /// Raised when a message is received.
+        /// </summary>
+        event Action<string, SignalingMessage> MessageReceived;
+    }
+
+    public class PeerSignalingService : ISignalingService
+    {
+        public IEnumerable<string> RemoteEndPoints => throw new NotImplementedException();
+
+        public event Action RemoteEndPointsChanged;
+        public event Action<string, SignalingMessage> MessageReceived;
+
+        public PeerSignalingService(IPAddress broadcast, ushort port, string category)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Dispose()
+        {
+            throw new NotImplementedException();
+        }
+
+        public void SendToRemoteEndpoint(string targetEndPointId, SignalingMessage message)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Start(string localEndPointId)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Stop()
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    static class SignalingExamples
+    {
+        // Use the service to connect a local PeerConnection to another chosen by the user among the available ones.
+        public static void ChooseAndConnect(PeerConnection pc)
```

Comment:
This looks like a useful utility, if taking an `ISignalingService` as parameter.

Reply:
Yes, will try to turn it into a utility.